### PR TITLE
docs: refine API docs for files in Command dir

### PR DESCRIPTION
**Pull request by devvaannsh**
The following changes are made:-

API Documentation Cleanup:
Removed all internal details from the API docs.
Now, only the exports.* content is included.

DefaultMenus.js Exclusion:
This file is internal, so it won’t show up in the API docs anymore.

Refactor in Menus.js:
The object literals inside MenuSection are now declared separately and assigned to the MenuSection object afterward.
This doesn’t affect any existing functionality.
The change was made because Docusaurus had trouble parsing nested object literals when generating markdown, which caused errors.

### CODE REVIEW

**Comment by devvaannsh** [comment]
The tests were failing because in file Menus.js, the variables were used before they were initialized! Fixed the issue!

File: src/command/CommandManager.js [hunk 1]
```diff
@@ -83,6 +104,7 @@ define(function (require, exports, module) {
 
     /**
      * Get command id
+     * @private
```
Comment (abose) [inline src/command/CommandManager.js:107]: The Command class is effectively public as an object is constructed by `register` API and returned. So we need to retain it as public. Modify all API usages below as public too wherever relevant.

**Review by abose — CHANGES_REQUESTED** [review]
We need to add the following docs too:


- [x] Commands.js - The well known exported command IDs needs to be in the docs somehow. maybe use enum?
- [x] KeyBindingManager.js
- [ ] KeyboardOverlayMode.js
- [x] Keys.js - maybe use enum?

Tracked in https://github.com/phcode-dev/phoenix/issues/1922

**Comment by devvaannsh** [comment]
In keybindingManager.js, the @private tags are placed after the function description. Why? Because the jsdoc-2-md library showed a warning that @private tags cannot have a description. So, if @private is defined above the description, it mistakenly treats the description as part of @private.

In keybindingManager.js, the @type tag for UN_SWALLOWED_EVENTS was changed from {(string)[]} to {Array.<string>}, because the former format is invalid and throws an error.

In Commands.js, each export needs to be handled separately. JSDoc should be added for each export individually so that they show up correctly in the API docs.

**Comment by abose** [comment]
LGTM. Thanks for contributing @devvaannsh

**Pull request merged**